File: Лаборатори/Lab3-Sort.py
class Solution:
    def sortArray(self, nums: List[int]) -> List[int]:
        
        def merge(left: List[int], right: List[int]) -> List[int]:
            sorted_array = []
            i = j = 0
            
            while i < len(left) and j < len(right):
                if left[i] <= right[j]:
                    sorted_array.append(left[i])
                    i += 1
                else:
                    sorted_array.append(right[j])
                    j += 1
            
            sorted_array.extend(left[i:])
            sorted_array.extend(right[j:])
            
            return sorted_array

        def merge_sort(nums: List[int]) -> List[int]:
            if len(nums) <= 1:
                return nums 
            
            mid = len(nums) // 2
            left_half = merge_sort(nums[:mid])
            right_half = merge_sort(nums[mid:])
            
            return merge(left_half, right_half)

        return merge_sort(nums)

File: Лаборатори/Lab3-Sort.java
class Solution {
    public int[] sortArray(int[] nums) {
        return mergeSort(nums);
    }

    private int[] mergeSort(int[] nums) {
        if (nums.length <= 1) {
            return nums;
        }

        int mid = nums.length / 2;
        int[] leftHalf = mergeSort(Arrays.copyOfRange(nums, 0, mid));
        int[] rightHalf = mergeSort(Arrays.copyOfRange(nums, mid, nums.length));

        return merge(leftHalf, rightHalf);
    }

    private int[] merge(int[] left, int[] right) {
        int[] sortedArray = new int[left.length + right.length];
        int i = 0, j = 0, k = 0;

        while (i < left.length && j < right.length) {
            if (left[i] <= right[j]) {
                sortedArray[k++] = left[i++];
            } else {
                sortedArray[k++] = right[j++];
            }
        }

        while (i < left.length) {
            sortedArray[k++] = left[i++];
        }
        while (j < right.length) {
            sortedArray[k++] = right[j++];
        }

        return sortedArray;
    }
}